allennlp/semparse/contexts/atis_tables.py

```diff
@@ -0,0 +1,437 @@
+          'date_day': ['month_number', 'day_number', 'year', 'day_name'],
+          'days': ['days_code', 'day_name'],
+          'equipment_sequence': ['aircraft_code_sequence', 'aircraft_code'],
+          'fare': ['fare_id', 'from_airport', 'to_airport', 'fare_basis_code',
+                   'fare_airline', 'restriction_code', 'one_direction_cost',
+                   'round_trip_cost', 'round_trip_required'],
+          'fare_basis': ['fare_basis_code', 'booking_class', 'class_type', 'premium', 'economy',
+                         'discounted', 'night', 'season', 'basis_days'],
+          'flight': ['flight_id', 'flight_days', 'from_airport', 'to_airport', 'departure_time',
+                     'arrival_time', 'airline_flight', 'airline_code', 'flight_number',
+                     'aircraft_code_sequence', 'meal_code', 'stops', 'connections',
+                     'dual_carrier', 'time_elapsed'],
+          'flight_fare': ['flight_id', 'fare_id'],
+          'flight_leg': ['flight_id', 'leg_number', 'leg_flight'],
+          'flight_stop': ['flight_id', 'stop_number', 'stop_days', 'stop_airport',
+                          'arrival_time', 'arrival_airline', 'arrival_flight_number',
+                          'departure_time', 'departure_airline', 'departure_flight_number',
+                          'stop_time'],
+          'food_service': ['meal_code', 'meal_number', 'compartment', 'meal_description'],
+          'ground_service': ['city_code', 'airport_code', 'transport_type', 'ground_fare'],
+          'month': ['month_number', 'month_name'],
+          'restriction': ['restriction_code', 'advance_purchase', 'stopovers',
+                          'saturday_stay_required', 'minimum_stay', 'maximum_stay',
+                          'application', 'no_discounts'],
+          'state': ['state_code', 'state_name', 'country_name']}
+
+YES_NO = {'one way': 'NO',
+          'economy': 'YES'}
+
+AIRPORT_CODES = ['ATL', 'NA', 'OS', 'UR', 'WI', 'CLE', 'CLT', 'CMH',
+                 'CVG', 'DAL', 'DCA', 'DEN', 'DET', 'DFW', 'DTW',
+                 'EWR', 'HOU', 'HPN', 'IAD', 'IAH', 'IND', 'JFK',
+                 'LAS', 'LAX', 'LGA', 'LG', 'MCI', 'MCO', 'MDW', 'MEM',
+                 'MIA', 'MKE', 'MSP', 'OAK', 'ONT', 'ORD', 'PHL', 'PHX',
+                 'PIE', 'PIT', 'SAN', 'SEA', 'SFO', 'SJC', 'SLC',
+                 'STL', 'TPA', 'YKZ', 'YMX', 'YTZ', 'YUL', 'YYZ']
+
+AIRLINE_CODE_LIST = ['AR', '3J', 'AC', '9X', 'ZW', 'AS', '7V',
+                     'AA', 'TZ', 'HP', 'DH', 'EV', 'BE', 'BA',
+                     'HQ', 'CP', 'KW', 'SX', '9L', 'OH', 'CO',
+                     'OK', 'DL', '9E', 'QD', 'LH', 'XJ', 'MG',
+                     'YX', 'NX', '2V', 'NW', 'RP', 'AT', 'SN',
+                     'OO', 'WN', 'TG', 'FF', '9N', 'TW', 'RZ',
+                     'UA', 'US', 'OE']
+
+CITIES = ['NASHVILLE', 'BOSTON', 'BURBANK', 'BALTIMORE', 'CHICAGO', 'CLEVELAND',
```

Comment:
I like this idea, it is kind of how I thought we might do it originally and I think it would make learning easier, but wasn't sure how we would go about implementing it. Based on what you suggested, I'm thinking we might do this at the binary expression level. Something like:

```biexpr = ( "city" ws "." ws "city_name"  binop ws city_strings ) / ...
city_strings = "NASHVILLE" \ "BOSTON" \  ...```

To keep it general we'd probably want to do it for all the columns, but practically that would create a ton of productions. Maybe something like automatically checking the training set for what columns appear in where clauses and constructing that grammar based on that could be in a future PR.